host/volume/api/http.go

```diff
@@ -0,0 +1,114 @@
+package volumeapi
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/flynn/flynn/Godeps/_workspace/src/github.com/julienschmidt/httprouter"
+	"github.com/flynn/flynn/host/volume"
+	"github.com/flynn/flynn/host/volume/zfs"
+	"github.com/flynn/flynn/pkg/httphelper"
+	"github.com/flynn/flynn/pkg/random"
+)
+
+type HttpAPI struct {
+	vman *volume.Manager
+}
+
+func NewHttpAPI(vman *volume.Manager) *HttpAPI {
+	return &HttpAPI{vman: vman}
+}
+
+func RegisterRoutes(api *HttpAPI, r *httprouter.Router) {
+	r.POST("/volume/provider", api.CreateProvider)
+	r.POST("/volume/provider/:provider_id/newVolume", api.Create)
+	r.PUT("/volume/instances/:id/snapshot", api.Snapshot)
+	r.GET("/volume/instances/:id/inspect", api.Inspect)
```

Comment:
Now that I think about it, I think a `/storage` namespace makes a lot of sense, since we already namespace other calls under `/host` and `/cluster`.